app/src/main/java/com/pvp/app/ui/screen/calendar/MonthlyCalendarScreen.kt

```diff
@@ -0,0 +1,240 @@
+fun MonthlyCalendar(
+    days: Array<String>,
+    month: YearMonth,
+    dates: List<CalendarUiState.DateEntry>,
+    onClickListener: (CalendarUiState.DateEntry) -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp)
+    ) {
+        Row {
+            repeat(days.size) {
+                val item = days[it]
+                DayItem(item, modifier = Modifier.weight(1f))
+            }
+        }
+
+        Header(
+            month = month,
+        )
+
+        Content(
+            dates = dates,
+            onClickListener = onClickListener
+        )
+    }
+}
+
+@Composable
+fun Header(
+    month: YearMonth,
+    model: MonthlyCalendarViewModel = hiltViewModel()
+) {
+    Row {
+        IconButton(onClick = {
+            model.changeMonth(month.minusMonths(1))
+        }) {
+            Icon(
+                imageVector = Icons.AutoMirrored.Filled.KeyboardArrowLeft,
+                contentDescription = "Back"
+            )
+        }
+
+        Text(
+            text = month.getDisplayName(),
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.bodyLarge,
+            modifier = Modifier
+                .weight(1f)
+                .align(Alignment.CenterVertically)
+        )
+
+        IconButton(onClick = {
+            model.changeMonth(month.plusMonths(1))
+        }) {
+            Icon(
+                imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
+                contentDescription = "Forward"
+            )
+        }
+    }
+}
+
+@Composable
+fun DayItem(day: String, modifier: Modifier = Modifier) {
+    Box(modifier = modifier) {
+        Text(
+            text = day,
+            style = MaterialTheme.typography.bodyMedium,
+            color = MaterialTheme.colorScheme.primary,
+            modifier = Modifier
+                .align(Alignment.Center)
+                .padding(10.dp)
+        )
+    }
+}
+
+@Composable
+fun Content(
+    dates: List<CalendarUiState.DateEntry>,
+    onClickListener: (CalendarUiState.DateEntry) -> Unit,
+) {
+    Column {
+        var index = 0
+
+        repeat(6) {
+            if (index >= dates.size) {
+                Log.e("CONTENT DATE FILTER", dates.size.toString())
+                return@repeat
+            }
+
+            Row {
+                repeat(7) {
+                    val item =
+                        if (index < dates.size) {
+                            dates[index]
+                        } else {
+                            CalendarUiState.DateEntry.Empty
+                        }
+
+                    ContentItem(
+                        date = item,
+                        onClickListener = onClickListener,
+                        modifier = Modifier.weight(1f)
+                    )
+
+                    index++
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ContentItem(
+    date: CalendarUiState.DateEntry,
```

Comment:
Could we name it `entry` instead? To still make it quite understandable to some extent and evade chained `date.date.<...>` calls that by name should return date but first one doesn't.